# HG changeset patch
# User alta88@fixall.com
# Date 1744228906 -7200
# Parent  2b30b4a5315be4ff5f74b081993acfa6c74d2877
Bug 1846550 - add ability to manually sort (order) folders in folder pane. Additional dnd changes by alta88.

Without preventDefault() in onDragOver, the default action is to cancel the onDrop. In addition to that, the dropEffect has to be set.
However, for no-ops (where onDragOver target is invalid), the default action has to happen to prevent, for example, the
external dnd noop from being left in a state where further dnd, onDragStart, is not called again.

So complex onDragOver decisions have to be very carefully structured. See:
https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/Drag_operations#droptargets

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -3129,8 +3129,9 @@ var folderPane = {
           // Insert before the target.
           this._clearDropTarget();
           row.classList.add("reorder-target-before");
           event.dataTransfer.dropEffect = "move";
+          event.preventDefault();
           return;
         }
         if (
           event.clientY > center + quarterOfHeight &&
@@ -3140,8 +3141,9 @@ var folderPane = {
           // Insert after the target.
           this._clearDropTarget();
           row.classList.add("reorder-target-after");
           event.dataTransfer.dropEffect = "move";
+          event.preventDefault();
           return;
         }
       }
 
